metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -159,4 +189,68 @@ private void publishToSolr(RdfWrapper rdf, FullBeanImpl fullBean) throws Indexin
       throw new RecordRelatedIndexingException(SOLR_SERVER_PUBLISH_ERROR, e);
     }
   }
+
+  private String searchMatchingRecordForRedirection(String recordIdIncludingDatasetId,
+      List<String> datasetIdsForRedirection)
+      throws IndexerRelatedIndexingException, RecordRelatedIndexingException {
+    //Check if record with same record identifier exists
+    final Map<String, String> queryParamMap = new HashMap<>();
+    queryParamMap.put("q",
+        String.format("%s:%s", EdmLabel.EUROPEANA_ID.toString(), recordIdIncludingDatasetId));
+    queryParamMap.put("fl", EdmLabel.EUROPEANA_ID.toString());
+
+    SolrDocumentList documents = getSolrDocuments(queryParamMap);
+    if (!CollectionUtils.isEmpty(documents)) {
+      return null;
+    }
+
+    //Check matches with older dataset identifiers
+    if (!CollectionUtils.isEmpty(datasetIdsForRedirection)) {
+      //The incoming structure of the identifier is /datasetId/recordId
+      final String[] splitRecordIdentifier = recordIdIncludingDatasetId.split("/");
+      String datasetId = splitRecordIdentifier[1];
```

Comment:
Apparently we don't need this variable. Is that right? The code does not use it.

Reply:
Correct, I was holding it for the next ticket, will remove it for now.